libecho/src/EchoProtocol.cc

```diff
@@ -0,0 +1,220 @@
+#include "EchoProtocol.h"
+#include <cassert>
+#include <QDebug>
+
+using namespace std::chrono_literals;
+
+static constexpr size_t PACKET_SIZE = 50;
+
+// NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
+EchoProtocol::EchoProtocol(int winsize, int send_freq, int recv_freq, int lim) {
+    big_win_size = static_cast<double>(winsize) / 44100 * (PACKET_SIZE + 10) * 12s;
+    buffer = new uint8_t[2 * PACKET_SIZE];
+    closed = false;
+    connection = std::unique_ptr<EchoRawConnection>(EchoRawConnection::getBitEchoRawConnection(
+        winsize, send_freq, recv_freq, lim));
+    thr[0] = thr[1] = nullptr;
+    is_connected = false;
+    status = 0;
+    lastPacketAcked = 0;
+    number = 0;
```

Comment:
Przerób ten konstruktor i klasę tak by używał initializer list oraz in-class member initialization